app/src/main/cpp/BrowserWorld.cpp

```diff
@@ -717,8 +717,8 @@ BrowserWorld::State::SortWidgets() {
     Widget* wa = da->second.first;
     Widget* wb = db->second.first;
 
-    // Parenting sort
-    if (wa && wb && wa->IsVisible() && wb->IsVisible()) {
+    // Parenting sort only in case both elements have the same normalized z value
+    if (wa && wb && wa->IsVisible() && wb->IsVisible() && da->second.second == db->second.second) {
```

Comment:
Since the depth is a `float` we should probably be checking if the two depth values are within a certain epsilon of each other and not use `==`. For example:
```
std::fabs(da->second.second - db->second.second) < FLT_EPSILON
```